Metro-Records/la-metro-councilmatic
# Add ability to manage alerts

## Overview

This branch adds ui to allow logged in users to manage alerts. There is a collapsible alert manager on the homepage where users can:
- select the type of alert
- see a small preview of the styling
- enter a description
- view and delete existing alerts

Closes #1010 

### Demo
![ezgif-1-fd05881540](https://github.com/Metro-Records/la-metro-councilmatic/assets/114717958/b6e41399-b6c9-41ed-ba0d-bfee35a8f2f5)

### Notes

At the time of writing, the Bootstrap 5 changes in #1009 have yet to be merged. This presents a couple caveats:

- The alert styles "primary" and "secondary" listed in the dropdown are present in bs5, but not the version of bootstrap that's used by this branch, so alerts with those types won't be styled as of now. Those styles will display properly once that pr comes in.
-  I have inline styles here that I will replace with their corresponding bs5 classes once that pr gets merged in.

## Testing Instructions

 * Bring the container up to run the new migration
 * Log in to the site and navigate to the home page
   * Confirm that the alert management section does not show on the home page until after you've logged in
 * Click the show button to expand the alert section and click through some of the alert type choices
   * Confirm that the select box style changes to preview what the alert will look like
 * Enter an alert description and click Submit
   * Confirm that the alert gets added to the top of every page, and to the View Existing section of the alert manager
 * Click the delete button next to your newly added alert in the alert manager
   * Confirm that the alert disappears from the top
